src/include/plan_node/export_external_file_plan_node.h

```diff
@@ -0,0 +1,157 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include "plan_node/abstract_plan_node.h"
+
+// TODO(Gus,Wen): This plan probably needs a different way of generating the output schema. The output schema should be
+// the childs output schema. But also maybe this node doesnt even need an output schema in the execution layer, so I put
+// it as null_ptr for now
+
+namespace terrier::plan_node {
+
+/**
+ * This is the plan node when exporting data from the database into an external
+ * file. It is configured with the name of the file to write content into, and
+ * the delimiter, quote, and escape characters to use when writing content.
+ */
+class ExportExternalFilePlanNode : public AbstractPlanNode {
+ protected:
+  /**
+   * Builder for a export external file scan plan node
+   */
+  class Builder : public AbstractPlanNode::Builder<Builder> {
+   public:
+    /**
+     * Dont allow builder to be copied or moved
+     */
+    DISALLOW_COPY_AND_MOVE(Builder);
+
+    /**
+     * @param file_name file path for external file file
+     * @return builder object
+     */
+    Builder &SetFileName(std::string file_name) {
+      file_name_ = std::move(file_name);
+      return *this;
+    }
+
+    /**
+     * @param delimiter delimiter character for external file
+     * @return builder object
+     */
+    Builder &SetDelimiter(char delimiter) {
+      delimiter_ = delimiter;
+      return *this;
+    }
+
+    /**
+     * @param quote quote character for external file
+     * @return builder object
+     */
+    Builder &SetQuote(char quote) {
+      quote_ = quote;
+      return *this;
+    }
+
+    /**
+     * @param escape escape character for external file
+     * @return builder object
+     */
+    Builder &SetEscape(char escape) {
+      escape_ = escape;
+      return *this;
+    }
+
+    /**
+     * Build the export external file scan plan node
+     * @return plan node
+     */
+    std::shared_ptr<ExportExternalFilePlanNode> Build() {
+      return std::shared_ptr<ExportExternalFilePlanNode>(
+          new ExportExternalFilePlanNode(std::move(children_), file_name_, delimiter_, quote_, escape_));
+    }
+
+   protected:
+    /**
+     * string representation of file name
+     */
+    std::string file_name_;
+    /**
+     * delimiter character
+     */
+    char delimiter_ = ',';
+    /**
+     * quote character
+     */
+    char quote_ = '"';
+    /**
+     * escape character
+     */
+    char escape_ = '"';
+  };
+
+  /**
+   * @param children child plan nodes
+   * @param file_name string representation of file name
+   * @param delimiter delimiter character
+   * @param quote quote character
+   * @param escape escape character
+   */
+  explicit ExportExternalFilePlanNode(std::vector<std::unique_ptr<AbstractPlanNode>> &&children, std::string file_name,
+                                      char delimiter = ',', char quote = '"', char escape = '\"')
```

Comment:
I say we get rid of default argument values.

@tli2 What does the design style say?

Reply:
I forgot to remove these. For default arguments I've been making then default values to the member variables of the respective builder class.